src/nvim/autocmd.c

```diff
@@ -2093,3 +2312,92 @@ bool au_exists(const char *const arg) FUNC_ATTR_WARN_UNUSED_RESULT
   xfree(arg_save);
   return retval;
 }
+
+static bool autocmd_should_be_removed(AutoCmd *ac)
+{
+  return ac->cmd == NULL;
+
+  AutocmdCommand command = ac->command;
+
+  switch (command.type) {
+    case COMMAND_EX:
+      return command.callable.c_cmd == NULL;
+    case COMMAND_VIML_FUNC:
+      ILOG("NOT IMPLEMENTED");
+      return autocmd_command_is_new(command);
+    case COMMAND_LUA:
+      return command.callable.c_lua == LUA_NOREF;
+    default:
+      return false;
+  }
+}
+
+static bool autocmd_command_is_new(AutocmdCommand command)
+{
+  switch (command.type) {
+    case COMMAND_EX:
+      return command.callable.c_cmd != NULL;
+    case COMMAND_VIML_FUNC:
+      ILOG("NOT IMPLEMENTED");
```

Comment:
is this planned to be implemented inside this PR?

Reply:
Yeah :)

Reply:
It will test if I did a good job of abstracting at the right points :)